### DISCUSSION TITLE
Tactics - Security 1

### DISCUSSION THREAD

**Discussion by changjurhee** [changjurhee/AONE#11] [Polls]
## Quality Attributes

Type | Description | QA Scenario (Stimulus / Source of stiulus / Artifact / Environment / Response / Response measure) | Tatics
 -- | -- | -- | --
Security | System shall handle the user’s password in secure way | Even if the server data is leaked to the outside by an insider, it must be encrypted and stored so that the password is not exposed. <br>- **Stimulus:** Data is leaked to the outside<br>- **Source:** Insider<br>- **Artifact:** User's password<br>- **Environment:**  All<br>- **Response:** Store user's password encrypted<br>- **Response measure:** Unknown raw user's password |

**Comment by BrunoYoo** [comment]
이 QA는 Server에 암호데이터를 저장한것이 유출되지 않도록 하는것이니, client에서 전달받은 암호를 저장할때, 단방향 Hash SHA-256 를 사용해서 Database에 암호 원본이 아닌 Hash data를 저장하면 될것 같습니다. Verify Message Integrity 항목의 SHA예가 있으나,  교제의 SHA 는 원본 미저장의 의미보다는 원본 변경의 검증에 가깝긴합니다. Encrypt data와 둘중에 혼란스럽긴 하지만 Encrypt data를 선정했습니다.

**Reply by changjurhee** [reply]
네, 의견 감사합니다. 투표도 해주세요. ^^;

**Comment by ch0115** [comment]
encrypt data 가 일반적인 것 같아 투표 했습니다.
